librad/src/git/types.rs

```diff
@@ -153,6 +153,118 @@ impl<'a> Into<ext::blob::Branch<'a>> for &'a Reference {
     }
 }
 
+pub struct SymbolicReference<'a> {
+    repo: &'a git2::Repository,
```

Comment:
I don't think this should take the `Repository`. We only need to know about a `Repository` when we create the symref. We should separate these concerns.